samples/ChatSample/Hubs/Chat.cs

```diff
@@ -5,31 +5,38 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.SignalR;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
 
 namespace ChatSample.Hubs
 {
-    // TODO: Make this work
     [Authorize]
     public class Chat : Hub
     {
+        private static readonly IDictionary<string, string> usersOnline = new ConcurrentDictionary<string, string>();
```

Comment:
Not threads, different servers that use the same scaleout backend. If server 1 gets a user then it gets added to server 1's collection, then server 2 gets a user and when you invoke SetUsersOnline server 2 doesn't have the user from server 1 so server 2's user doesn't know that server 1's user is online.